src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -48,9 +48,11 @@ public class VsPackageInstaller : IVsPackageInstaller2
             _sourceRepositoryProvider = sourceRepositoryProvider;
             _settings = settings;
             _solutionManager = solutionManager;
-            _projectContext = new VSAPIProjectContext();
             _packageServices = packageServices;
             _deleteOnRestartManager = deleteOnRestartManager;
+
+            _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
```

Comment:
Either way, as part of your validation of this change I would highly suggest you add a `ThreadHelper.ThrowIfNotOnUIThread()` to your `VSAPIProjectContext` constructor so that if anyone is still calling it on a background thread you'll know about it right away. Perhaps even throw in a Debug.Assert and try out some scenarios in case any of those exceptions get swallowed.
You *might* choose to not add that Throw method in to the actual PR since adding such method calls is great but not in escrow. But it would still help validate that your change here is reasonable safe if the alarm never goes off.